### DISCUSSION TITLE
The recommended way of logging?

### DISCUSSION THREAD

**Discussion by 0xthk** [blitz-js/blitz#2600] [Q&A] [answered]
Hi, 

I just start using blitz for my next project. After browsing the document, I can't find a section about logger something like this https://redwoodjs.com/docs/logger.

Is there a recommended way to handle logging on Blitz? Or just choose a logging framework like `pino` and integrate myself?

Thanks!

**Comment by flybayer** [comment, marked as answer]
Good question! We will be adding equivalent to what Redwood has. Internally we use https://github.com/fullstack-build/tslog, and so in my Blitz apps I'm also using tslog (importing and creating a logger manually).

**Reply by 0xthk** [reply]
Thanks for the answer! I will take a look at `tslog`. I am using `pino` currently. 😀

**Reply by kdasme** [reply]
@flybayer `tslog` looks great, is there any way to access and/or configure main Blitz tslog logger (e.g. pass `{ type: "json" }`)? Ideally, I just want to be able to filter logs related to each request (via request ID passed from the reverse proxy / nginx as HTTP header). :) Poor man's tracing, in other words.

**Reply by kdasme** [reply]
Answered the question myself - currently not possible to configure that ([here](https://github.com/blitz-js/blitz/blob/canary/nextjs/packages/next/server/lib/logging.ts#L41)), I'll try to dedicate time to create a PR to add more configuration properties (including `type`) as production environments require structured logs.

**Reply by flybayer** [reply]
yes, not yet, but we should support that!
